src/api/spec/controllers/concerns/triggerable_spec.rb

```diff
@@ -0,0 +1,173 @@
+require 'rails_helper'
+
+RSpec.describe Triggerable do
+  let(:fakes_controller) do
+    Class.new(ApplicationController) do
+      include Triggerable
+    end
+  end
+
+  let(:object) { fakes_controller.new }
+
+  let(:user) { create(:confirmed_user, login: 'foo') }
+  let(:project) { create(:project, name: 'project', maintainer: user) }
+  let(:package) { create(:package, name: 'package_trigger', project: project) }
+  let(:repository) { create(:repository, name: 'package_test_repository', architectures: ['x86_64'], project: project) }
+
+  before do
+    object.instance_variable_set(:@token, token)
```

Comment:
if you have a method `token` (even being private) on the object, you can work with `allow` and `receive`  too

Reply:
but `token` is an instance variable, with the `allow` and `receive` it does not seem to be possible to mock instance variables